test/metabase/models/secret/keystore_test.clj

```diff
@@ -0,0 +1,78 @@
+(ns metabase.models.secret.keystore-test
+  (:require [clojure.java.io :as io]
+            [clojure.test :refer :all]
+            [metabase.api.common :as api]
+            [metabase.models :refer [Database Secret]]
+            [metabase.test :as mt]
+            [metabase.test.fixtures :as fixtures])
+  (:import [java.io ByteArrayOutputStream File]
+           java.nio.charset.StandardCharsets
+           [java.security KeyStore KeyStore$PasswordProtection KeyStore$SecretKeyEntry]
+           javax.crypto.SecretKey
+           javax.crypto.spec.SecretKeySpec))
+
+(use-fixtures :once (fixtures/initialize :db :plugins :test-drivers))
+
+(defn create-test-jks-instance
+  "Creates a JKS instance and stores the given `entries` in it, using the given `ks-password` as the store password.
+  `entries` is a map from entry names to private key values. The KeyStore will be in PKCS12 format, for easier storage
+  and retrieval of arbitrary secret key data. Individual entries will be stored with password protection using
+  `ks-password` (a bad idea in production, but fine for tests)."
+  {:added "0.41.0"}
+  ^KeyStore [^String ks-password entries]
+  (let [pw         (.toCharArray ks-password)
+        protection (KeyStore$PasswordProtection. pw)]
+    (let [^KeyStore key-store (doto (KeyStore/getInstance "PKCS12")
```

Comment:
nit: no reason for this extra let. could just be in the parent let

Reply:
Good point; will fix it.

Reply:
Fixed